hottbox/algorithms/decomposition/fusion/cmtf.py

```diff
@@ -0,0 +1,210 @@
+from hottbox.core.operations import khatri_rao, hadamard
+from hottbox.utils.generation.basic import super_diag_tensor
+
+
+# TODO: Organise this better - lazy work around used
+class CMTF(BaseCPD):
+    """ Coupled Matrix and Tensor factorization for two ``tensors`` of order n and 2
+        with respect to a specified ``rank``. Computed via alternating least squares (ALS)
+    Parameters
+    ----------
+    max_iter : int
+        Maximum number of iteration
+    epsilon : float
+        Threshold for the relative error of approximation.
+    tol : float
+        Threshold for convergence of factor matrices
+    random_state : int
+    verbose : bool
+        If True, enable verbose output
+    Attributes
+    ----------
+    cost : list
+        A list of relative approximation errors at each iteration of the algorithm.
+
+    References
+    ----------
+
+    ..  [1] Acar, Evrim, Evangelos E. Papalexakis, Gozde Gurdeniz, Morten A. Rasmussen,
+            Anders J. Lawaetz, Mathias Nilsson and Rasmus Bro.
+            “Structure-revealing data fusion.” BMC Bioinformatics (2013).
+    ..  [2] Jeon, Byungsoo & Jeon, Inah & Sael, Lee & Kang, U. (2016).
+            SCouT: Scalable coupled matrix-tensor factorization—Algorithm and discoveries.
+            Int. Conf. Data Eng.. 811-822. 10.1109/ICDE.2016.7498292.
+    """
+    # TODO: change init use requiring a change in TensorCPD
+    def __init__(self, max_iter=50, epsilon=10e-3, tol=10e-5,
+                 random_state=None, verbose=False) -> None:
+        super(CMTF, self).__init__(init='random',
+                                   max_iter=max_iter,
+                                   epsilon=epsilon,
+                                   tol=tol,
+                                   random_state=random_state,
+                                   verbose=verbose)
+        self.cost = []
+
+    def copy(self):
+        """ Copy of the CPD algorithm as a new object """
+        new_object = super(CMTF, self).copy()
+        new_object.cost = []
+        return new_object
+
+    @property
+    def name(self):
+        """ Name of the decomposition
+        Returns
+        -------
+        decomposition_name : str
+        """
+        decomposition_name = super(CMTF, self).name
+        return decomposition_name
+
+    def decompose(self, tensor, mlst, rank):
+        """ Performs factorisation using ALS on the two instances of ``tensor``
+            with respect to the specified ``rank``
+
+        Parameters
+        ----------
+        tensor : Tensor
+            Multi-dimensional data to be decomposed
+        mlst : List of `Tensor`
+            List of two-dimensional `Tensor` to be decomposed
+        rank : tuple
+            Desired Kruskal rank for the given ``tensor``. Should contain only one value.
+            If it is greater then any of dimensions then random initialisation is used
+        Returns
+        -------
+        tensor_cpd : TensorCPD
```

Comment:
Docstring of what is returned doesn't correspond to what is actually returned. Also `Notes` section can be removed since it is for CPD-ALS.

Reply:
Completely missed this, thanks!